From 3e542e7eb2b98c9440965ff374cef2708efa0d82 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 12 Jan 2023 15:47:20 +0100
Subject: [PATCH] install python component

---
 CMakeLists.txt | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b42b33..7bf61df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -390,8 +390,18 @@ endif()
 # pybind11 module
 option(BUILDPYTHONMODULE "Build Python binding." OFF)
 if (BUILDPYTHONMODULE)
+  find_package(
+    Python3
+    COMPONENTS Interpreter Development.Module
+    REQUIRED
+  )
   find_package(pybind11 REQUIRED)
   pybind11_add_module(kahip_python_binding ${CMAKE_CURRENT_SOURCE_DIR}/misc/pymodule/kahip.cpp)
   target_link_libraries(kahip_python_binding PUBLIC kahip_static)
   set_target_properties(kahip_python_binding PROPERTIES OUTPUT_NAME "kahip")
+  install(TARGETS kahip_python_binding
+    LIBRARY DESTINATION "${Python3_SITELIB}"
+    COMPONENT python
+    EXCLUDE_FROM_ALL
+  )
 endif ()
-- 
2.42.0

