From 27e3817890b8b99fbe83ba024fef4b8fae1eba4d Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Fri, 25 Apr 2025 19:58:41 +0200
Subject: [PATCH 1/2] install Python lib to site-packages as its own component

---
 CMakeLists.txt | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3fa7c7a..4eca71a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -397,6 +397,11 @@ endif()
 # pybind11 module
 option(BUILDPYTHONMODULE "Build Python binding." OFF)
 if (BUILDPYTHONMODULE)
+  find_package(
+    Python3
+    COMPONENTS Interpreter Development.Module
+    REQUIRED
+  )
   find_package(pybind11 REQUIRED)
   pybind11_add_module(kahip_python_binding ${CMAKE_CURRENT_SOURCE_DIR}/misc/pymodule/kahip.cpp)
 
@@ -406,7 +411,11 @@ if (BUILDPYTHONMODULE)
     install(TARGETS kahip_python_binding DESTINATION kahip)
   else()
     # Standard installation path
-    install(TARGETS kahip_python_binding DESTINATION "${CMAKE_INSTALL_PYTHONDIR}/kahip")
+    install(TARGETS kahip_python_binding
+      LIBRARY DESTINATION "${Python3_SITELIB}"
+      COMPONENT python
+      EXCLUDE_FROM_ALL
+    )
   endif()
 
   target_link_libraries(kahip_python_binding PUBLIC kahip_static)
-- 
2.49.0

