From 81192b752cd5769c98c4cc2ea02400a1ffdc1332 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 12 Jan 2023 15:47:20 +0100
Subject: [PATCH] install python compoment

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b42b33..5a21e1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -394,4 +394,9 @@ if (BUILDPYTHONMODULE)
   pybind11_add_module(kahip_python_binding ${CMAKE_CURRENT_SOURCE_DIR}/misc/pymodule/kahip.cpp)
   target_link_libraries(kahip_python_binding PUBLIC kahip_static)
   set_target_properties(kahip_python_binding PROPERTIES OUTPUT_NAME "kahip")
+  install(TARGETS kahip_python_binding
+    LIBRARY DESTINATION "${Python3_SITELIB}"
+    COMPONENT python
+    EXCLUDE_FROM_ALL
+  )
 endif ()
-- 
2.42.0

